gstd/src/prog/salt_factory.rs

```diff
@@ -0,0 +1,35 @@
+pub struct SaltFactory([u8; 32]);
+
+fn invert(array: &mut [u8]) {
```

Comment:
What is the profit of inversion?

Reply:
I also thought about rotation, reversion, and appending itself. First, two can keep salt unchanged, and the last one makes allocations. Invert simple, and fast and the result always differs. Maybe you knew better operations?